src/vs/workbench/browser/labels.ts

```diff
@@ -218,8 +218,8 @@ export class FileLabel extends ResourceLabel {
 
 		this.setLabel({
 			resource,
-			name: (options && options.hideLabel) ? void 0 : paths.basename(resource.fsPath),
-			description: !hidePath ? getPathLabel(paths.dirname(resource.fsPath), rootProvider, this.environmentService) : void 0
+			name: (options && options.hideLabel) ? void 0 : paths.basename(resource.toString()),
+			description: !hidePath ? getPathLabel(paths.dirname(resource.toString()), rootProvider, this.environmentService) : void 0
```

Comment:
@isidorn Better check for `path` and use `authority` otherwise

Reply:
@jrieken will do, I am checking if it is better to introduce a brand new `FtpLabel`, instead of putting everything into `FileLabel`

Reply:
👎 Nah, this will become hairy I'd say. This isn't only about FTP but about all kind of remote sources. So, I'd say that the `FileLabel` is a good place to bundle the smartness.